src/lib/operators/print.cpp

```diff
@@ -152,4 +164,64 @@ std::string Print::_truncate_cell(const AllTypeVariant& cell, uint16_t max_width
   return cell_string;
 }
 
+std::string Print::_segment_type(const std::shared_ptr<BaseSegment>& segment) const {
+  std::string segment_type;
+  segment_type.reserve(8);
+  segment_type += "<";
+
+  if (std::dynamic_pointer_cast<BaseValueSegment>(segment)) {
+    segment_type += "ValueS";
+  } else if (std::dynamic_pointer_cast<ReferenceSegment>(segment)) {
+    segment_type += "ReferS";
+  } else if (const auto& encoded_segment = std::dynamic_pointer_cast<BaseEncodedSegment>(segment)) {
+    switch (encoded_segment->encoding_type()) {
+      case EncodingType::Unencoded: {
+        Fail("An actual segment should never have this type");
+      }
+      case EncodingType::Dictionary: {
+        segment_type += "Dic:";
+        break;
+      }
+      case EncodingType::RunLength: {
+        segment_type += "RLE:";
+        break;
+      }
+      case EncodingType::FixedStringDictionary: {
+        segment_type += "FSD:";
+        break;
+      }
+      case EncodingType::FrameOfReference: {
+        segment_type += "FoR:";
+        break;
+      }
+    }
+    if (encoded_segment->compressed_vector_type()) {
+      switch (*encoded_segment->compressed_vector_type()) {
+        case CompressedVectorType::FixedSize4ByteAligned: {
+          segment_type += "4B";
+          break;
+        }
+        case CompressedVectorType::FixedSize2ByteAligned: {
+          segment_type += "2B";
+          break;
+        }
+        case CompressedVectorType::FixedSize1ByteAligned: {
+          segment_type += "1B";
+          break;
+        }
+        case CompressedVectorType::SimdBp128: {
+          segment_type += "BP";
+          break;
+        }
+      }
+    } else {
+      segment_type += "--";
```

Comment:
That would print `RLE:--` for RLE-encoded columns, right? Why not simply `RLE` and only append for encoding types that have the ability to further compress?